modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/rest/SW360AttachmentAwareClient.java

```diff
@@ -62,14 +69,18 @@ public T uploadAndAttachAttachment(T itemToModify, Path fileToAttach, String kin
 
     private T uploadAndAttachAttachment(T itemToModify, Path fileToAttach, HttpEntity<MultiValueMap<String, Object>> requestEntity) {
         final String self = itemToModify.get_Links().getSelf().getHref();
-        ResponseEntity<T> response = restTemplate.postForEntity(self + ATTACHMENTS_ENDPOINT, requestEntity, getHandledClassType());
+        try {
+            ResponseEntity<T> response = restTemplate.postForEntity(self + ATTACHMENTS_ENDPOINT, requestEntity, getHandledClassType());
 
-        if (response.getStatusCode().is2xxSuccessful()) {
-            return Optional.ofNullable(response.getBody())
-                    .orElseThrow(() -> new ExecutionException("Body was null"));
-        } else {
-            throw new ExecutionException("Request to get attach " + fileToAttach + " to " + self + " failed with "
-                    + response.getStatusCode());
+            checkRestStatus(response);
+            return response.getBody();
```

Comment:
We should not return null here, use the getSaveOrThrow with Identical method as second parameter